contracts/pools/PoolCollection.sol

```diff
@@ -647,7 +675,22 @@ contract PoolCollection is IPoolCollection, Owned, BlockNumber, Utils {
         greaterThanZero(poolTokenAmount)
         returns (WithdrawalAmounts memory)
     {
-        InternalWithdrawalAmounts memory amounts = _poolWithdrawalAmounts(pool, _poolStorage(pool), poolTokenAmount);
+        Pool storage data = _poolData[pool];
```

Comment:
Some fields in `data.liquidity` (like `stakedBalance` and `baseTokenTradingLiquidity`) are being read from storage several times in this context (i.e., in this function and in the function called from this function).

Probably better to read `data.liquidity` into memory once, then use it in this function and in the other function (especially because that other function takes it by `memory`, which means that the entire `data` object is read from storage into memory before that function is called anyway).

In fact, that other function requires only `data.liquidity` and `data. tradingFeePPM` from that object, so probably better to just pass those two fields.